resources/assets/coffee/_classes/sticky-header.coffee

```diff
@@ -16,27 +16,60 @@
 #    along with osu!web.  If not, see <http://www.gnu.org/licenses/>.
 ###
 
+# Attachment that shows up below the omni-header.
 # How to use:
-# 1. create a marker on when it should be fixed, with class including
-#    'js-sticky-header' and data attribute 'data-sticky-header-target'
-# 2. subscribe to 'stickyHeader' event
-# 3. in the function, check if second parameter (first one is unused event
-#    object) is the correct target
-# 4. stick if matches, unstick otherwise
+# 1. render content into 'js-sticky-header-content' and 'js-sticky-header-breadcrumbs'
+# 2. Add 'js-sticky-header' class to a marker element that should cause the sticky to show.
 class @StickyHeader
+  @breadcrumbsElement: ->
+    document.getElementById('js-sticky-header-breadcrumbs')
+
+
+  @contentElement: ->
+    document.getElementById('js-sticky-header-content')
+
+
+  @headerHeight: ->
+    styles = window._styles.header
+    if osu.isMobile()
+      styles.heightMobile
+    else
+      styles.heightSticky
+
+
   constructor: ->
     @stickMarker = document.getElementsByClassName('js-sticky-header')
+    @visible = false
 
+    $(window).on 'throttled-scroll', @applyCss
     $(window).on 'throttled-scroll throttled-resize', @stickOrUnstick
     $(document).on 'turbolinks:load osu:page:change', @stickOrUnstick
 
 
+  applyCss: ->
+    header = document.getElementById('js-pinned-header')
```

Comment:
my guess says initializing/storing `document.getElementsByClassName` and using it like `@header[0]` is slightly faster.